src/server/scripts/Outland/CoilfangReservoir/TheUnderbog/boss_hungarfen.cpp

```diff
@@ -159,8 +157,36 @@ struct npc_underbog_mushroom : public ScriptedAI
     uint32 _counter;
 };
 
+// 31693 - Putrid Mushroom Primer
+class spell_hungarfen_putrid_mushroom_primer : public SpellScript
+{
+    PrepareSpellScript(spell_hungarfen_putrid_mushroom_primer);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_SUMMON_UNDERBOG_MUSHROOM });
+    }
+
+    void HandleScript(SpellEffIndex /*effIndex*/)
+    {
+        Creature* caster = GetCaster()->ToCreature();
+        if (!caster || !caster->IsAIEnabled())
+            return;
+
+        // This is wrong. They don't need a script to simply select random target. But what should be done here? FilterTargets? Exclude pets?
+        if (Unit* target = caster->AI()->SelectTarget(SelectTargetMethod::Random, 0))
```

Comment:
what's the target type for this spell? I'm assuming it's an AOE chooser since it's got a radius of 100 yards on it, so the correct way would be filtering through all of those targets if you want to exclude any, Spell system on itself already randomly chooses between targets when using AOE types, even when MaxTargets is set.

Reply:
I can add
```cpp
    void FilterTargets(std::list<WorldObject*>& targets)
    {
        if (targets.empty())
            return;

        WorldObject* target = Trinity::Containers::SelectRandomContainerElement(targets);
        targets.clear();
        targets.push_back(target);
    }

Reply:
Yes, a target script is the preferred way to handle situations like this

Does not apply to this spell (because its serverside) but for spells visible on client doing random selection in effect hook would play spell visual on every unit

Reply:
I can simply filter targets but it's still unknown what exactly that script should do, simply selecting random target still looks too easy

Reply:
since you lack information, the only way to test this is forcing that spell several times until you find a pattern, if you can't find any, random it is, for as long as you can reproduce the expected result

Reply:
I think better to check sniffs but so far I don't have any sniff

Reply:
is the .empty check really necessary at the beginning? if there are no targets available it shouldn't even trigger OnObjectAreaTargetSelect iirc

Reply:
> is the .empty check really necessary at the beginning? if there are no targets available it shouldn't even trigger OnObjectAreaTargetSelect iirc, it would make sense if it was after some type of filtering in case you set it empty entirely

YES

Reply:
if there are no targets available, why would the system force the hook processing at all?